src/Illuminate/Database/Schema/Grammars/MySqlGrammar.php

```diff
@@ -638,6 +638,10 @@ protected function modifyDefault(Blueprint $blueprint, Fluent $column)
         if (! is_null($column->default)) {
             return ' default '.$this->getDefaultValue($column->default);
         }
+
+        if (! $column->nullable && ! $column->useCurrent && in_array($column->type, ['timestamp', 'timestampTz'])) {
```

Comment:
This could be done in the timestamp and timestampTz methods. It would simplify the branch statement eg:

    protected function typeTimestamp(Fluent $column)
    {
        if ($column->useCurrent) {
            return 'timestamp default CURRENT_TIMESTAMP';
        }

        if (! $column->nullable) {
            return 'timestamp default 0';
        }

        return 'timestamp';
    }